argoverse/evaluation/stereo/model_analysis.py

```diff
@@ -0,0 +1,131 @@
+from pathlib import Path
+from typing import Dict, List, Optional, Tuple, Union
+
+import numpy as np
+import torch
+from fvcore.nn import ActivationCountAnalysis, FlopCountAnalysis, flop_count_table
+from rich.progress import track
+
+
+def model_analysis(
+    model: torch.nn.Module,
+    inputs: Union[torch.Tensor, Tuple[torch.Tensor, ...]],
+    output_dir: Path,
+) -> None:
+    """Compute number of parametes, number of activations, flops, and latency (inference time) of a given model.
+
+    Args:
+        model: The model to perform the analysis on.
+        inputs: Inputs that are passed to the model.
+        output_dir: The output directory to save the final txt report with the analysis.
+    """
+    flops = FlopCountAnalysis(model, inputs)
+    activations = ActivationCountAnalysis(model, inputs)
+
+    count_table = flop_count_table(flops=flops, activations=activations, max_depth=3)
+    time_stats = compute_forward_time(model, inputs)
+
+    input_size_str = "\n".join(f"Input size {i}: {input.shape}" for i, input in enumerate(inputs))
+    output_str = (
+        f"Device name: {torch.cuda.get_device_name(0)} \n"
```

Comment:
Might be good to warn the user if they're not using a GPU. Maybe check if cuda is available at the top of the function?